charts/quickwit/templates/_helpers.tpl

```diff
@@ -190,20 +200,35 @@ Quickwit metastore environment
 - name: QW_METASTORE_URI
   value: {{ .Values.config.metastore_uri }}
 {{- else if .Values.config.postgres }}
+- name: QW_METASTORE_URI
+  value: "postgres://$(POSTGRES_USERNAME):$(POSTGRES_PASSWORD)@$(POSTGRES_HOST):$(POSTGRES_PORT)/$(POSTGRES_DATABASE)"      
+{{- end }}
+{{- if (.Values.config.postgres).host }}
 - name: POSTGRES_HOST
   value: {{ required "A valid config.postgres.host is required!" .Values.config.postgres.host }}
+{{- end }}
+{{- if (.Values.config.postgres).port }}
 - name: POSTGRES_PORT
   value: {{ .Values.config.postgres.port | default 5432 | quote }}
+{{- end }}
+{{- if (.Values.config.postgres).database }}
 - name: POSTGRES_DATABASE
   value: {{ .Values.config.postgres.database | default "metastore" }}
+{{- end }}
+{{- if (.Values.config.postgres).username }}
 - name: POSTGRES_USERNAME
   value: {{ .Values.config.postgres.username | default "quickwit" }}
+{{- end }}
+{{- if (.Values.config.postgres).password_secret_key_ref }}
+- name: POSTGRES_PASSWORD
+  valueFrom:
+    secretKeyRef:
+      {{- toYaml .Values.config.postgres.password_secret_key_ref | nindent 6 }}
+{{- else if (.Values.config.postgres).password }}
 - name: POSTGRES_PASSWORD
   valueFrom:
     secretKeyRef:
       name: {{ include "quickwit.fullname" . }}
```

Comment:
we should keep the existing logic here:
```
if .Values.config.postgres
  - (.Values.config.postgres).host is a required value
  - port, database and username are set with a default value
```

Only the password should change according to the presense of `(.Values.config.postgres).password_secret_key_ref` or `(.Values.config.postgres).password`